File: finanze/domain/use_cases/search_crypto_assets.py
import abc

from domain.crypto import AvailableCryptoAssetsRequest, AvailableCryptoAssetsResult


class SearchCryptoAssets(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    async def execute(
        self, request: AvailableCryptoAssetsRequest
    ) -> AvailableCryptoAssetsResult:
        raise NotImplementedError

File: mobile/src/domain/usecases/core/searchCryptoAssets.ts
import {
  AvailableCryptoAssetsRequest,
  AvailableCryptoAssetsResult,
} from "@/domain"

export interface SearchCryptoAssets {
  execute(
    request: AvailableCryptoAssetsRequest,
  ): Promise<AvailableCryptoAssetsResult>
}
